lib/elixir/lib/gen_event.ex

```diff
@@ -655,11 +655,12 @@ defmodule GenEvent do
   Record.defrecordp :handler, [:module, :id, :state, :pid, :ref]
 
   @doc false
-  def init_it(starter, :self, name, mod, args, options) do
+  def init_it(starter, :self, name, mod, args, options)
+      when is_atom(mod) and is_list(args) and is_list(options) do
     init_it(starter, self(), name, mod, args, options)
   end
 
-  def init_it(starter, parent, name, _, _, options) do
+  def init_it(starter, parent, name, _mod, _args, options) when is_list(options) do
```

Comment:
I am unsure about `handlers` here. We don't need to check that `options` is a list for the same reason as above.